File: shared/src/main/java/ru/privatenull/pnauth/api/AuthUser.java
package ru.privatenull.pnauth.api;

import java.time.Instant;
import java.util.UUID;

public record AuthUser(
        UUID uniqueId,
        String username,
        Instant registeredAt,
        Instant lastLoginAt,
        boolean premium,
        boolean totpEnabled,
        String lastIp,
        DialogPreference dialogPreference
) {
    public AuthUser(UUID uniqueId, String username, Instant registeredAt, Instant lastLoginAt) {
        this(uniqueId, username, registeredAt, lastLoginAt, false, false, null, DialogPreference.AUTO);
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthUser.kt
package ru.privatenull.pnauth.api

import java.time.Instant
import java.util.UUID

@JvmRecord
data class AuthUser @JvmOverloads constructor(
    val uniqueId: UUID,
    val username: String,
    val registeredAt: Instant?,
    val lastLoginAt: Instant?,
    val premium: Boolean = false,
    val totpEnabled: Boolean = false,
    val lastIp: String? = null,
    val dialogPreference: DialogPreference = DialogPreference.AUTO
)

